docs/en/sdk-manual/migrate-from-third-party-sdks.md

```diff
@@ -0,0 +1,122 @@
+
+```javascript
+import { ClickstreamAnalytics } from "@aws/clickstream-web";
+import { getAnalytics, logEvent, setUserProperties, setUserId } from "firebase/analytics";
+
+export const AnalyticsLogger = {
+
+  log(eventName, attributes, items) {
+    attributes = attributes ?? {}
+    const {["items"]: items, ...mAttributes} = attributes;
+    
+    // Clickstream SDK
+    ClickstreamAnalytics.record({
+      name: eventName,
+      attributes: mAttributes,
+      items: items
+    })
+
+    //Firebase SDK
+    const analytics = getAnalytics();
+    logEvent(analytics, eventName, attributes);
+  },
+
+  setUserAttributes(attributes) {
+    // Clickstream SDK
+    ClickstreamAnalytics.setUserAttributes(attributes);
+
+    // Firebase SDK
+    const analytics = getAnalytics();
+    setUserProperties(analytics, attributes);
+  },
+
+  setUserId(userId) {
+    //Clickstream SDK
+    ClickstreamAnalytics.setUserId(userId)
+
+    //Firebase SDK
+    const analytics = getAnalytics();
+    setUserId(analytics, userId);
+  },
+}
+```
+
+We need to encapsulate three APIs `log()` 、`setUserAttributes()` and `setUserId()`, that's all. When we invoke the `AnalyticsLogger.log('testEvent')`  method, both Clickstream and Firebase SDK will log the event, so we only need to call the `AnalyticsLogger` api in your business code.
+
+## Step 3: Migrate to common APIs in minutes
+
+### For log events
+
+```javascript
+  onSignedUp(user) {
+    let attributes = {
+      _user_id: user.id,
+      username: user.username,
+      email: user.email,
+    };
+--  logEvent(analytics, 'sign_up', attributes);
+++  AnalyticsLogger.log('sign_up', attributes);
+  }
+```
+
+For the events log api, we just need to get the event name, attributes and pass them into the new api. Of course, you can also use the "Replace in File" feature to make quick changes, as shown in the image below.
+
+![replace_in_files](../images/sdk-manual/replace-in-file.png) 
+
+### For user
+
+```javascript
+  userSignedIn(user) {
+--  setUserId(analytics, user.id);
+++  AnalyticsLogger.setUserId(user.id);
+    let attributes = {
+      _user_id: user.id,
+      username: user.username,
+      email: user.email,
+    };
+--  setUserProperties(analytics, attributes);
+++  AnalyticsLogger.setUserAttributes(attributes);
+  }
+```
+
+For user id, replace  `setUserId()` with `AnalyticsLogger.setUserId()` .
+
+For user attributes, replace `setUserProperties()` with `AnalyticsLogger.setUserAttributes()` .
+
+## Summary
+
+As we saw above, it is easy to get Clickstream SDK and Firebase SDK to work together, after these three steps, your data will both upload to Clickstream analytics and Firebase, these two SDKs will work well together, and has no influence with each other, after your analytics business is migrated to clickstream, you only need to modify the `AnalyticsLogger` file to smoothly remove another SDK, and save your costs.
```

Comment:
As we saw above, it is easy to get Clickstream SDK and Firebase SDK to work together, after these three steps, your data will both upload to Clickstream analytics and Firebase, these two SDKs will work well together, and has no influence with each other. After you satisfy with the data, you only need to modify the `AnalyticsLogger` file to smoothly remove or disable another SDK.

Reply:
changed to this